src/lib/dlt_user.c

```diff
@@ -4169,13 +4189,25 @@ DltReturnValue dlt_user_log_send_log(DltContextData *log, const int mtype, int *
             /* check trace load before output */
             if (!sent_size)
             {
+                if ((uint32_t)log->handle->log_level_pos > dlt_user.dlt_ll_ts_num_entries) {
+                    char msg[255];
+                    sprintf(msg, "log handle has invalid log level pos %d, current entries: %u, dropping message\n",
+                            log->handle->log_level_pos, dlt_user.dlt_ll_ts_num_entries);
+                    dlt_user_output_internal_msg(LOG_ERR, msg, NULL);
+                    return DLT_RETURN_ERROR;
+                }
+
                 pthread_rwlock_wrlock(&trace_load_rw_lock);
-                DltTraceLoadSettings* settings =
-                    dlt_find_runtime_trace_load_settings(
+                dlt_ll_ts_type* ll_ts = &dlt_user.dlt_ll_ts[log->handle->log_level_pos];
+                if (ll_ts->trace_load_settings == NULL) {
+                    ll_ts->trace_load_settings = dlt_find_runtime_trace_load_settings(
```

Comment:
The trace load settings should be check null before .

Reply:
Can you elaborate on this? Null check for tl settings is above and this if is only executed when it is null

Reply:
Do you mean checking ll_ts for null? I think that check would be superfluous

Reply:
After carefully checking, it is superfluous to check  ll_ts->trace_load_settings null. Because this pointer already check null and print log inside func "dlt_check_trace_load"